bokehjs/src/lib/models/layouts/layout_dom.ts

```diff
@@ -433,13 +435,34 @@ export abstract class LayoutDOMView extends UIElementView {
     this._after_layout()
   }
 
-  override render_to(element: Node): void {
-    element.appendChild(this.el)
-    this.build()
+  private _was_built: boolean = false
+  override render_to(element: Node | null): void {
+    if (!this.is_layout_root)
+      throw new Error(`${this.toString()} is not a root layout`)
+
+    this.render()
+    if (element != null)
+      element.appendChild(this.el)
+    this.r_after_render()
+    this._was_built = true
+
     this.notify_finished()
   }
 
+  r_after_render(): void {
```

Comment:
What is the "r_" for?

Reply:
Recursive. I will likely eliminate this method in another round of changes.